### PR TITLE
DIS-1648 Add linting

### PR DESCRIPTION
**Pull request by dbirks**
This should bring no functional changes, only styling/formatting changes.

- Add a `.eslintrc.js` that extends `index.js` (to use the common configuration).
- Add husky. Looks like they've changed the way they set things up with v7. (A new `.husky` folder, and a `prepare` npm script.)
- Add Github Action to lint.
- Add the required devDependencies. This includes some typescript-eslint packages that we'll be able to remove once we do the cleanup planned in [DIS-1647](https://eoilreports.atlassian.net/browse/DIS-1647).

I did some quick tests with hpapi and aurora-service with this `2.1.0-beta.0`, and didn't see any difference in the linting warnings/errors.

Before merge:
- [x] Bump version to 2.1.0

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+module.exports = {
+  extends: ["./index.js"],
+  // Disabling some rules for index.js due to some strange typescript errors that should be fixed in DIS-1647
```
Comment (babennettdev) [inline .eslintrc.js:3]: 👍

File: .github/workflows/code-checks.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        uses: actions/setup-node@v2
+        with:
+          registry-url: https://npm.pkg.github.com/
+          scope: "@polarislabs"
+      - name: Install dependencies
+        run: npm install
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NODE_AUTH_TOKEN }}
+      - name: Lint
+        run: npm run lint
```
Comment (babennettdev) [inline .github/workflows/code-checks.yml:29]: Linting was performed on this PR and succeeded 👍

File: package.json [hunk 3]
```diff
@@ -1,10 +1,13 @@
...
-  "version": "2.0.2",
+  "version": "2.1.0-beta.0",
   "description": "private ESLint package to use across polaris dev projects to maintain same standard of code.",
   "main": "index.js",
   "scripts": {
-    "deploy": "npm publish"
+    "deploy": "npm publish",
+    "lint": "eslint .",
+    "lint:fix": "eslint --fix .",
+    "prepare": "husky install"
```
Comment (babennettdev) [inline package.json:10]: What does this script do for us? I don't see where we call it, husky calls it, or our Github actions call it. Was it just used to install husky or is it used behind the scenes somewhere in the CI pipeline? If it is not used, could we remove it?

File: package.json [hunk 4]
```diff
@@ -1,10 +1,13 @@
 {
   "name": "@polarislabs/eslint-config-polaris",
-  "version": "2.0.2",
+  "version": "2.1.0-beta.0",
```
Comment (babennettdev) [inline package.json:3]: Don't forget to remove the `-beta` tag before merge 👀

**Review by babennettdev — CHANGES_REQUESTED** [review]
Who lints the linter 👀 ? Looks like our linter lints itself 🥇 !

Great work, David. Left a couple questions, but the linting has cleaned up the repo, our Github action is working successfully, and the changes look good!

File: package.json (hunk 3 above)
Comment (dbirks) [reply package.json:10]: Yes husky set that up, and I had to check the docs to double-check, but it looks like `prepare` is a reserved script name for commands you want to run on every `npm install`:
https://docs.npmjs.com/cli/v6/using-npm/scripts#life-cycle-scripts

File: package.json (hunk 4 above)
Comment (dbirks) [reply package.json:3]: Updating now...

File: package.json (hunk 3 above)
Comment (babennettdev) [reply package.json:10]: Awesome! Thank you for the clarification!

**Review by babennettdev — APPROVED** [review]
LGTM, David! Thank you for clarifying on the `prepare` npm script!

**Review by codetelo — APPROVED** [review]
Looks great! Let's get this in!

**Pull request merged**